### ISSUE TITLE
"needless borrow" for ISO_8859_1.decode does not compile

### ISSUE THREAD

**Issue by Maikel1963** [rust-lang/rust-clippy#8811]
In this code 
```rs
  let result = match File::open(filename) {
    Ok(mut f) => {
      let mut content =  Vec::new();
      match f.read_to_end(&mut content) {
        Ok(x) if x>0 => Some(ISO_8859_1.decode(**&content**, DecoderTrap::Replace).unwrap()),
        _ => {println!("Hier gibt's: [{}] \r\n als Inhalt.", String::from_utf8_lossy(&content)); None},

      }
    }
```
clippy suggest to remove the & from "&content" in the "OK" arm.
But this does not compile.

**Comment by Jarcho** [comment]
I can't reproduce that using:
```rust
use std::{fs::File, io::Read};
use encoding::{DecoderTrap, Encoding, all::ISO_8859_1};

fn main() {
	let result = match File::open("") {
		Ok(mut f) => {
			let mut content = Vec::new();
			match f.read_to_end(&mut content) {
				Ok(x) if x>0 => Some(ISO_8859_1.decode(&content, DecoderTrap::Replace).unwrap()),
				_ => {
					println!("Hier gibt's: [{}] \r\n als Inhalt.", String::from_utf8_lossy(&content));
					None
				},
			}
		}
		_ => panic!(),
    };
}
```

What version of clippy are you using?

**Comment by Maikel1963** [comment]
I have clippy version 0.1.60
that's strange:
I took your code and started to rewrite my function. And now clippy does not complain.
But I cannot spot any differences in the source code. 

Thanks for your help.
